build.gradle

```diff
@@ -1,46 +1,150 @@
 plugins {
-    id 'com.github.eerohele.dita-ot-gradle' version '0.4.2'
+    id 'com.github.eerohele.dita-ot-gradle' version '0.4.3'
+    id 'com.github.eerohele.saxon-gradle' version '0.3.0'
 }
 
-ditaOt.dir System.getProperty('dita.home', '../')
+import com.github.eerohele.DitaOtTask
+import com.github.eerohele.SaxonXsltTask
 
-import com.github.eerohele.AntBuilderAssistant
-def antBuilder = AntBuilderAssistant.getAntBuilder(project)
+def getPropertyOrDefault(String name, def defaultValue) {
+    hasProperty(name) ? findProperty(name) : defaultValue
+}
 
-task pdf {
-  doLast {
-    antBuilder.execute {
-      ant(antfile: 'build.xml', target: 'pdf')
+String ditaHome = getPropertyOrDefault('ditaHome', '../')
+String ditaHomeSrc = getPropertyOrDefault('ditaHomeSrc', ditaHome)
+String configDir = "${ditaHomeSrc}/config"
+String ditavalFile = "${projectDir}/platform.ditaval"
+Boolean toolkitBuild = file("${projectDir}/../lib/dost.jar").exists()
+String samplesDir = toolkitBuild ? "${ditaHome}/docsrc/samples" : "${projectDir}/samples"
+String outputDir = getPropertyOrDefault('outputDir', toolkitBuild ? "${ditaHome}/doc" : "${projectDir}/out")
+
+String toURI(String path) {
+    file(path).toURI().toString()
+}
+
+ditaOt.dir ditaHome
+
+task messages(type: SaxonXsltTask) {
+    input "${configDir}/messages.xml"
+    output 'user-guide/DITA-messages.xml'
```

Comment:
This and others assume the current repository is the docs repo or `docsrc` inside DITA-OT. This should be configurable with a paragraph, so that we can run Gradle from any random directory.

Reply:
I guess you mean that you should be able to do something like `gradle -b ~/Git/docs/build.gradle -PditaHome=$DITA_HOME site`, too, which you currently can't?

If so, I agree, and I pushed the changes.